modAionImpl/test/org/aion/zero/impl/AionHubTest.java

```diff
@@ -0,0 +1,209 @@
+import org.aion.mcf.trie.TrieImpl;
+import org.aion.zero.impl.config.CfgAion;
+import org.aion.zero.impl.db.AionRepositoryImpl;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.types.AionTransaction;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AionHubTest {
+
+    private void checkHubNullity(AionHub hub) {
+        assertThat(hub).isNotNull();
+        assertThat(hub.getBlockchain()).isNotNull();
+        assertThat(hub.getPendingState()).isNotNull();
+        assertThat(hub.getP2pMgr()).isNotNull();
+        assertThat(hub.getRepository()).isNotNull();
+        assertThat(hub.getEventMgr()).isNotNull();
+        assertThat(hub.getSyncMgr()).isNotNull();
+        assertThat(hub.getBlockStore()).isNotNull();
+        assertThat(hub.getPropHandler()).isNotNull();
+    }
+
+    @BeforeClass
+    public static void setup() {
+        // logging to see errors
+        Map<String, String> cfg = new HashMap<>();
+        cfg.put("GEN", "INFO");
+        cfg.put("CONS", "INFO");
+        cfg.put("DB", "ERROR");
+
+        AionLoggerFactory.init(cfg);
+    }
+
+    @After
+    public void exitAndKillThreads() {
+        boolean wait = true;
+
+        while (wait) {
+            wait = false;
+            try {
+                sleep(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            for (Thread t : Thread.getAllStackTraces().keySet()) {
+                if (t.isAlive()) {
+                    String name = t.getName();
+
+                    if (name.contains("p2p-") || name.contains("pool-") || name.contains("EpPS")) {
+                        t.interrupt();
+                        wait = true;
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    public void MockHubInst_wStartAtGenesis() {
+        StandaloneBlockchain.Builder builder = new StandaloneBlockchain.Builder();
+        StandaloneBlockchain.Bundle bundle = builder.withValidatorConfiguration("simple").build();
+
+        StandaloneBlockchain chain = bundle.bc;
+        chain.setBestBlock(chain.getGenesis());
+
+        CfgAion.inst().setGenesis(chain.getGenesis());
+
+        AionHub hub = AionHub.createForTesting(CfgAion.inst(), chain, chain.getRepository());
+        checkHubNullity(hub);
+
+        AionBlock blk = hub.getStartingBlock();
+        assertThat(blk).isNotNull();
+        assertThat(blk.getNumber()).isEqualTo(0);
+
+        hub.close();
+        assertThat(hub.isRunning()).isFalse();
+    }
+
+    @Test
+    public void MockHubInst_wStartAtBlock() {
+        StandaloneBlockchain.Builder builder = new StandaloneBlockchain.Builder();
+        StandaloneBlockchain.Bundle bundle = builder.withValidatorConfiguration("simple").build();
+
+        StandaloneBlockchain chain = bundle.bc;
+        int expectedStartBlock = 6;
+        generateRandomChainWithoutTransactions(chain, expectedStartBlock, 1);
```

Comment:
If this test fails in Jenkins, do we have any way of recovering what input data made it fail?  If not, would be nice to log it somehow.